llama_hub/llama_packs/raptor/base.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+from llama_index import Document, VectorStoreIndex
+from llama_index.bridge.pydantic import BaseModel, Field
+from llama_index.llama_pack.base import BaseLlamaPack
+from llama_index.llms import OpenAI
+from llama_index.llms.llm import LLM
+from llama_index.llms.types import ChatMessage, MessageRole
+from llama_index.prompts import PromptTemplate
+from sklearn.mixture import GaussianMixture
+from umap import UMAP
+
+SUMMARY_PROMPT_STR = """
+<system> You are a Summarizing Text Portal
+<user> Write a summary of the following, including as many key details as
+possible: {context}:
+<system>
```

Comment:
just curious -- are the system tags needed here? Normally this would just be a `system` message in `llm.chat()` I think?

Reply:
I missed this, it should be removed in the last update